lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2913,49 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok)) {
+    ParseRelativeBoundsClause(Result.get());
+  }
+
   return Result;
 }
 
+bool Parser::ParseRelativeBoundsClause(Expr *Expr) {
+  RelativeBoundsClause *Result;
+  ExprResult Results;
+  tok::TokenKind TokKind = Tok.getKind();
+  if (TokKind != tok::identifier) {
+    Diag(Tok, diag::err_expected_bounds_expr);
```

Comment:
I don't think this case can ever occur because the caller checks that the token is an identifier that starts a relative bounds clause.  I think you should replace the Diag message with an assert.